lib/spack/spack/package.py

```diff
@@ -898,6 +898,25 @@ def version(self):
                              " does not have a concrete version.")
         return self.spec.versions[0]
 
+    @property
+    def platform_executables(self):
+        # Executables on Windows end in
+        # a .exe extension, need to capture
+        # for regex
+        def to_windows_exe(exe: str):
+            if exe.endswith('$'):
```

Comment:
Three things:

* Every string ends with `$`, doesn't it? This should just append `.exe` if it isn't there already
* This is very similar to `test.cmd.external.platform_executables`: you can cut down on that by
  1. make a `_platform_executables` helper function that takes an extra `suffix` arg
  2. make a `platform_executables` function here that sets `.exe`
  3. now the implementation in `test.cmd.external` can be simplified: just call `_platform_executables(..., '.bat')`
* This function should move to `DetectablePackageMeta`